controller/server_test.go

```diff
@@ -261,3 +261,98 @@ func TestServer_TaskDelete(t *testing.T) {
 		})
 	}
 }
+
+func TestServer_TaskUpdate(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+	ctrl := ReadWrite{
+		baseController: &baseController{
+			conf:    &config.Config{},
+			drivers: driver.NewDrivers(),
+			logger:  logging.NewNullLogger(),
+		},
+		store: event.NewStore(),
+	}
+	ctrl.conf.Finalize()
+
+	t.Run("disable-then-enable", func(t *testing.T) {
+		taskName := "task_a"
+		taskConf := config.TaskConfig{
+			Name:   config.String("task_a"),
+			Module: config.String("findkim/print/cts"),
+			Condition: &config.ServicesConditionConfig{
+				ServicesMonitorConfig: config.ServicesMonitorConfig{
+					Names: []string{"service"},
+				},
+			},
+		}
+		taskConf.Finalize(ctrl.conf.BufferPeriod, *ctrl.conf.WorkingDir)
+		task, err := newDriverTask(ctrl.conf, &taskConf, nil)
+		require.NoError(t, err)
+
+		d := new(mocksD.Driver)
+		d.On("Task").Return(task).
+			On("UpdateTask", mock.Anything, driver.PatchTask{Enabled: false}).
+			Return(driver.InspectPlan{ChangesPresent: false, Plan: ""}, nil)
+		err = ctrl.drivers.Add(taskName, d)
+		require.NoError(t, err)
+
+		// Set the task to disabled
+		updateConf := config.TaskConfig{
+			Name:    &taskName,
+			Enabled: config.Bool(false),
+		}
+
+		changed, plan, _, err := ctrl.TaskUpdate(ctx, updateConf, "")
+		require.NoError(t, err)
+		assert.False(t, changed)
+		assert.Empty(t, plan)
+
+		// Re-enable the task
+		updateConf.Enabled = config.Bool(true)
+		d.On("UpdateTask", mock.Anything, driver.PatchTask{Enabled: true}).
+			Return(driver.InspectPlan{ChangesPresent: false, Plan: ""}, nil)
+		_, plan, _, err = ctrl.TaskUpdate(ctx, updateConf, "")
+		require.NoError(t, err)
+		assert.Empty(t, plan)
+
+		// No events since the task did not run
+		events := ctrl.store.Read(taskName)
+		assert.Empty(t, events)
```

Comment:
I liked that you checked for events here. Suggestion to add the events check to the other test cases since events implementation and testing was moved out of the api package.

Reply:
Sounds good, added!